### PR TITLE
Test related to children

### PR DESCRIPTION
**Pull request by DavidBruant**
PR for #14

### CODE REVIEW

**Comment by DavidBruant** [comment]
I love the :white_check_mark: from Travis on the last commit 

:tada:

File: bouture.js [hunk 1]
```diff
@@ -1,61 +1,98 @@
...
 
-tagNames.forEach(tag => {
-  Object.defineProperty(Bouture, tag, {
-    get: () => {
-      const element = document.createElement(tag)
+tagNames.forEach(tagName => {
+  Object.defineProperty(Bouture, tagName, {
+    get: function () {
+      let tags = [{name: tagName, args: []}]
```
Comment (DavidBruant) [inline bouture.js]: the name `args` is a bit low-level. Maybe something like `contents` would be more explicit on the intention

File: bouture.js [hunk 2]
```diff
@@ -1,61 +1,98 @@
...
-                    }
-                  })
-              }
-              break
-            default:
-              // Other types not handle by cases : Symbol, boolean, undefined
-          }
-        })
+        const lastTag = tags.pop()
+        tags.push({name: lastTag.name, args: args})
```
Comment (DavidBruant) [inline bouture.js]: The `pop` + `push` feels a bit wasteful (removing an element and adding it back in)
What about mutating the last element?

File: bouture.js [hunk 3]
```diff
@@ -1,61 +1,98 @@
...
       }
-      branche.getElement = () => {
-        return element
-      }
+
+      Object.defineProperty(branche, 'getElement', {
+        value: function () {
+          let elements = []
+          tags.forEach((tag, index) => {
+            if (!index) {
```
Comment (DavidBruant) [inline bouture.js]: Let's be more explicit here with `index === 0`

File: bouture.js [hunk 4]
```diff
@@ -1,61 +1,98 @@
...
+              elements = Bouture.completeElement(tag.name, tag.args)
+            } else {
+              elements.append(Bouture.completeElement(tag.name, tag.args))
```
Comment (DavidBruant) [inline bouture.js]: `Bouture.completeElement(tag.name, tag.args)` can be factored out in a variable

File: bouture.js [hunk 5]
```diff
@@ -1,61 +1,98 @@
-const Bouture = {}
+const Bouture = {
+  completeElement: (tag, args) => {
```
Comment (DavidBruant) [inline bouture.js]: The `completeElement` function is internal to the library, there is no reason to expose it as `Bouture.completeElement`. Let's make it module-private

Comment (pointbar) [reply bouture.js]: I've removing the method -completeElement- from Bouture, because I think the coupling is weak. But, I've a doubt.

**Pull request merged**